torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -2153,6 +2153,21 @@ c10::intrusive_ptr<Work> ProcessGroupGloo::allgather_coalesced(
   return work;
 }
 
+c10::intrusive_ptr<Work> ProcessGroupGloo::allgather_into_tensor_coalesced(
+    std::vector<at::Tensor>& outputs,
+    std::vector<at::Tensor>& inputs,
+    const AllgatherOptions& opts) {
+  TORCH_CHECK_EQ(outputs.size(), inputs.size());
+  std::vector<std::vector<at::Tensor>> output_lists(getSize());
+  for (auto& output : outputs) {
```

Comment:
nit: I think this can be `const`.
```suggestion
  for (const auto& output : outputs) {
```

Reply:
probably cannot do that because of std::move

Reply:
It looks like `chunks` would be a `vector` of new `Tensor` objects, so the `std::move` should not affect the original `output` tensor?
https://github.com/pytorch/pytorch/blob/65efbf078ced58240edc4abefff3cc9afc2b8a1f/aten/src/ATen/native/TensorShape.cpp#L925
I may be confused though 😭